test/pipelines/test_pipeline_yaml.py

```diff
@@ -1108,6 +1108,42 @@ def run_batch(self):
     assert pipeline.components["DummyNode"].non_alphanumeric_param == "\\[ümlaut\\]"
 
 
+def test_load_yaml_param_that_equals_component_name(tmp_path):
```

Comment:
Test only `Pipeline._load_or_get_component()`, not the whole yaml loading.

Reply:
Not sure if that's a good measure here. Generally testing "private" methods is not advisable. Are there any reasons why we should do this here?

Reply:
True that, but then we should test `Pipeline.load_from_config()` as that's the only public function that calls `Pipeline._load_or_get_component()`.

I still think it would be better to test only `Pipeline._load_or_get_component()`. The `Pipeline` is pretty complicated in itself and it doesn't worry me that we're testing private functions really. Specially when they're as complex as this one.

We already have some pretty chunky unit tests that require to declare this or that temporary component to verify everything is working properly, and I think we must keep them at bay.

I'd love if we could manage to keep tests as small as possible, doing the bare minimum to verify that something is working properly.

Reply:
Also let's remember to use `@pytest.mark.unit`.